communication/src/dtls_message_channel.cpp

```diff
@@ -455,6 +456,18 @@ ProtocolError DTLSMessageChannel::receive(Message& message)
 	return NO_ERROR;
 }
 
+/**
+ * Once data has been successfully received we can stop
+ * sending move-session messages.
+ * This is also used to reset the expiration counter.
+ */
+void DTLSMessageChannel::cancel_move_session()
+{
+    move_session = false;
+    sessionPersist.clear_use_count();
+    command(SAVE_SESSION);
```

Comment:
I'd add a check here and avoid invoking the `SAVE_SESSION` command if `use_counter` is already 0. The channel implementation calls `cancel_move_session()` on every message it receives